package/boot/grub2/Makefile

```
@@ -71,51 +76,73 @@ HOST_CONFIGURE_ARGS += \
 	--sbindir="$(STAGING_DIR_HOST)/bin" \
 	--disable-werror \
 	--disable-libzfs \
-	--disable-nls
+	--disable-nls \
+	--with-platform=none
 
 HOST_MAKE_FLAGS += \
 	TARGET_RANLIB=$(TARGET_RANLIB) \
 	LIBLZMA=$(STAGING_DIR_HOST)/lib/liblzma.a
 
+TARGET_CFLAGS :=
```

@uxgood I've missed this one and this seems wrong, why is this needed? I've tried to build without this change, but the build fails.

---

Maybe it's related to those macOS `x86_64-openwrt-linux-musl/bin/ld: libgrubkern.a(libgrubkern_a-misc.o): relocation R_X86_64_32S against .rodata' can not be used when making a PIE object; recompile with -fPIE` issues?

---

Topmake passed some options to package  by TARGET_CFLAGS, it makes grub2 fail to generate modules.So, I clear TARGET_CFLAGS in grub2 package.

---

-fno-plt in TARGET_CFLAGS makes module missing _GLOBAL_OFFSET_TABLE_

```suggestion
TARGET_CFLAGS += -fplt
```
```suggestion
TARGET_CFLAGS := $(filter-out -fno-plt,$(TARGET_CFLAGS)) 
```

---

I think it's a combination of things, possibly not just/only macos.  To get this to build I need to disable ASLR building along with the -fno-plt filtering:

PKG_ASLR_PIE:=0
TARGET_CFLAGS := $(filter-out -fno-plt,$(TARGET_CFLAGS))

---

> PKG_ASLR_PIE:=0

So you're building with `CONFIG_PKG_ASLR_PIE_ALL=y` ?

---

> TARGET_CFLAGS := $(filter-out -fno-plt,$(TARGET_CFLAGS)) 

@uxgood thanks, that looks good, makes sense and works fine, are you going to create some PR with the fixes? You've also had some other fix in the works for that `grub-bios-setup` error message during the boot of the `pc` variant

---

> > PKG_ASLR_PIE:=0
> 
> @ldir So you're building with `CONFIG_PKG_ASLR_PIE_ALL=y` ?

It turns out, yes I am.  I thought there was a commit that defaulted x86_64 to enable at some point.  Maybe it's by accident I've got this setting.  But it builds now and it boots too :-)

---

@ynezz Let me have a rest. I may do these things in next weeks. If anyone else is willing to do these things, I appreciate it

---

@uxgood no worries, fixed by @ldir-EDB0 via fcd14017007db35a4a41ef4fd6b69a5e02edbd5e